tests/unit/client_factory_test.py

```
@@ -0,0 +1,251 @@
+        assert client.operator_account_id is None
+        assert client.operator_private_key is None
+        client.close()
+
+
+class TestClientFromEnv:
+    """Tests for Client.from_env() class method."""
+
+    def test_from_env_missing_operator_id_raises_error(self):
+        """Test that from_env raises ValueError when OPERATOR_ID is missing."""
+        with patch.dict(os.environ, {"OPERATOR_KEY": "302e020100300506032b65700422042012345678901234567890123456789012"}, clear=True):
+            with pytest.raises(ValueError) as exc_info:
```

Ruff (0.14.10)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)